0_From_Code_First_To_First_Principles/1_code_first/assignment/src/UserModel.ts

```diff
@@ -0,0 +1,68 @@
+import { PrismaClient } from '@prisma/client';
+import { generateRandomPassword } from './utils/generateRandomPassword';
+
+const prisma = new PrismaClient();
```

Comment:
As to your question about where to instantiate the PrismaClient, I think it's generally wiser to separate the db implementation from our business logic, so our infrastructure won't be tightly coupled to our implementation (which makes it easier to mock and change db if necessary).

To implement this correctly I would create a db-service that implements all the relevant methods we use with Prisma (and do nothing else but call the Prisma api). Then I would call this service in this class.

However in this case, since you're already using Prisma directly inside these business logic methods and instantiating it in the same file, it's already tightly coupled, so I would instantiate it as part of the class.

Passing the Prisma instance through a constructor doesn't make sense here because that means you would have to instantiate it inside the userController, and that breaks the separation of concerns - userController uses the userModel (aham, service), which uses the db service (prisma). They're not supposed to know about each other (which again makes it easier for us to change/scale).
I guess the answer the AI gave here is the recommended approach in general for separating concerns, but in this case it will create more coupling.